great_expectations/rule_based_profiler/config/base.py

```diff
@@ -0,0 +1,214 @@
+    class Meta:
+        unknown = INCLUDE
+
+    __config__ = ParameterBuilderConfig
+
+    name = fields.String(required=True)
+    class_name = fields.String(required=True)
+    module_name = fields.String(
+        required=False,
+        all_none=True,
+        missing="great_expectations.rule_based_profiler.parameter_builder",
+    )
+    batch_request = fields.Dict(keys=fields.String(), required=False, allow_none=True)
+
+
+@dataclass(frozen=True)
+class ExpectationConfigurationBuilderConfig(DictDot):
+    expectation_type: str
+    class_name: str
+    module_name: Optional[str] = None
+    mostly: Optional[float] = None
+    meta: Optional[Dict] = None
+
+
+class ExpectationConfigurationBuilderConfigSchema(NotNullSchema):
+    class Meta:
+        unknown = INCLUDE
+
+    __config__ = ExpectationConfigurationBuilderConfig
+
+    class_name = fields.String(required=True)
+    module_name = fields.String(
+        required=False,
+        all_none=True,
+        missing="great_expectations.rule_based_profiler.expectation_configuration_builder",
+    )
+    expectation_type = fields.String(required=True)
+    mostly = fields.Float(required=False, allow_none=True)
+    meta = fields.Dict(required=False, allow_none=True)
+
+
+@dataclass(frozen=True)
+class RuleConfig(DictDot):
+    name: str
+    domain_builder: DomainBuilderConfig
+    parameter_builders: List[ParameterBuilderConfig]
+    expectation_configuration_builders: List[ExpectationConfigurationBuilderConfig]
+
+
+class RuleConfigSchema(NotNullSchema):
+    class Meta:
+        unknown = INCLUDE
+
+    __config__ = RuleConfig
+
+    name = fields.String(required=True)
+    domain_builder = fields.Nested(DomainBuilderConfigSchema, required=True)
+    parameter_builders = fields.List(
+        cls_or_instance=fields.Nested(ParameterBuilderConfigSchema, required=True),
+        required=True,
+    )
+    expectation_configuration_builders = fields.List(
+        cls_or_instance=fields.Nested(
+            ExpectationConfigurationBuilderConfigSchema, required=True
+        ),
+        required=True,
+    )
+
+
+@dataclass
+class RuleBasedProfilerConfig(BaseYamlConfig):
+    name: str
+    config_version: float
+    rules: Dict[str, RuleConfig]
+    variables: Optional[Dict[str, Any]] = None
+    commented_map: Optional[CommentedMap] = None
+
+    def __post_init__(self):
+        # Required to fully set up the commented map and enable serialization
+        super().__init__()
+
+    @classmethod
+    def get_config_class(cls) -> Type["RuleBasedProfilerConfig"]:
+        return cls
+
+    @classmethod
+    def get_schema_class(cls) -> Type["RuleBasedProfilerConfigSchema"]:
+        return RuleBasedProfilerConfigSchema
+
+
+class RuleBasedProfilerConfigSchema(NotNullSchema):
+    class Meta:
+        unknown = INCLUDE
+
+    __config__ = RuleBasedProfilerConfig
+
+    name = fields.String(required=True)
+    config_version = fields.Float(required=True)
```

Comment:
@cdkini Do we need `if_missing` here?  Could you please check against what we did in `Checkpoint` and do something reasonable?  Thanks.

Reply:
I've updated the `config_version` to have validation per the `CheckpointConfigSchema`. I don't think we should have a default value if missing since we're requiring a config version per the design doc (let me know if this should be changed)